src/core/meshcop/dataset_manager.cpp

```diff
@@ -772,6 +772,10 @@ void PendingDatasetManager::HandleDelayTimer(void)
 {
     DelayTimerTlv *delayTimer;
     Dataset        dataset;
+    Dataset        aDataset;
+    Tlv           *sTlv;
+    Tlv           *eTlv;
+    Tlv           *pTlv;
```

Comment:
Move definitions under `#if` block or we get warning for unused variables (and warnings are treated as error).